packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,34 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // reset to first page if tabbing back into the carousel after previously moving pages
+    if (currentContains && currentCardIndex === 0) {
+      this.start = 0;
+    }
+
+    if (currentContains) {
+      // going forwards, change page depending on card index
+      if (currentCardIndex >= this.start + this.pageSize) {
+        this.start = Math.min(currentCardIndex, this._total + 1 - this.pageSize);
+
+        // going backwards, change page depending on card index
+      } else if (currentCardIndex < this.start) {
+        this.start = currentCardIndex + 1 - this.pageSize;
```

Comment:
```suggestion
        this.start = Math.max(currentCardIndex + 1 - this.pageSize, 0);
```

Also, this logic conflicts with your logic for "previous focus was outside carousel and now on carousel" scenario. I suggest using `event.relatedTarget` to detect that and do things appropriately here.